packages/design-system/vite.config.ts

```diff
@@ -26,7 +46,41 @@ export default defineConfig(() => {
     build: {
       sourcemap: true,
       rollupOptions: {
-        external: [...Object.keys(pkg.peerDependencies), "react/jsx-runtime"],
+        output: {
+          entryFileNames: "[name].[hash].js",
+          chunkFileNames: "[name].[hash].js",
+          assetFileNames: "[name].[hash][extname]",
+          preserveModules: true,
+        },
+        external: (id) => {
```

Comment:
Excludes all dependencies from the bundle and throws an error if a devDependency is accessed from the bundle.